src/Bicep.Core/Analyzers/Linter/Rules/NoUnusedExistingResourcesRule.cs

```diff
@@ -35,7 +36,8 @@ override public IEnumerable<IDiagnostic> AnalyzeInternal(SemanticModel model)
             var unreferencedResources = model.Root.Declarations.OfType<ResourceSymbol>()
                 .Where(sym => sym.NameSyntax.IsValid)
                 .Where(sym => sym.DeclaringResource.IsExistingResource())
-                .Where(sym => !model.FindReferences(sym).Any(rf => rf != sym.DeclaringResource));
+                .Where(sym => !model.FindReferences(sym).Any(rf => rf != sym.DeclaringResource))
+                .Where(sym => (sym.DeclaringResource.TryGetBody()?.Resources ?? Enumerable.Empty<ResourceDeclarationSyntax>()).All(child => child.IsExistingResource()));
```

Comment:
@anthony-c-martin I've added a check for child resources.

I'm also testing if the child resources are existing resources. However is this logically possible, or an unnecessary check?

Reply:
Yes it's definitely possible to have an existing resource with nested existing resources.

I would probably keep it simple here and just say that if there are **any** nested children (`existing` or not), then the parent resource is being referenced. This would mean that in the case where there are children (but they are all unused), we wouldn't show diagnostics on the parent resource. This seems perfectly acceptable to me - this linter will pick up the individual children; if the user chooses to remove them, then the linter will pick up the parent resource.

The alternative (if being fully accurate) would be to recursively check whether each child (and their respective children/grandchildren etc) is being referenced, and show this diagnostic only if there are no references to any of the resources in this tree. This feels like a niche case to me, and not worth adding complexity to solve.